sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/plans/logical/object.scala

```diff
@@ -44,18 +53,30 @@ trait ObjectOperator extends LogicalPlan {
   def withObjectOutput: LogicalPlan = if (output.head.dataType.isInstanceOf[ObjectType]) {
     this
   } else {
-    withNewSerializer(outputObject)
+    withNewSerializer(outputObject :: Nil)
   }
 
   /** Returns a copy of this operator with a different serializer. */
-  def withNewSerializer(newSerializer: NamedExpression): LogicalPlan = makeCopy {
+  def withNewSerializer(newSerializer: Seq[NamedExpression]): LogicalPlan = makeCopy {
```

Comment:
according to the definition of serializer: `def serializer: Seq[NamedExpression]`, a seq of named expressions work together as a serialize, so `withNewSerializer` seems more accuracy here?